File: Leetcode/Daily/27-05-2024/prob.go
func lower_bound(arr []int, ele int)int{
    l,r := 0,len(arr)

    for l<r{
        mid := (l+r)/2
        if arr[mid]>=ele{
            r = mid
        }else{
            l = mid+1
        }
    }

    return l
}

func specialArray(nums []int) int {
    sort.Ints(nums)
    n := len(nums)
    maxi := nums[n-1]
    for i:=1;i<=maxi;i++{
        lb := lower_bound(nums,i)
        if n-lb == i{
            return i
        }
    }

    return -1
}

File: Leetcode/Daily/27-05-2024/prob.cpp
class Solution {
public:
    int specialArray(vector<int>& nums) {

        int n = nums.size();
        sort(nums.begin(),nums.end());

        int maxi = nums[nums.size()-1];
        for(int i=1;i<=maxi;i++){
            int lb = lower_bound(nums.begin(),nums.end(),i)-nums.begin();
            if(n-lb == i){
                return i;
            }
        }

        return -1;
    }
};